### DISCUSSION TITLE
🍨 Add JPEGView to Scoop

### DISCUSSION THREAD

**Discussion by pureby** [sylikc/jpegview#110] [Ideas]
Hello Kevin! First of all I want to thank you for reviving this incredible app! When it comes to image viewers, nothing else comes _even close_ to JPEGView — even the old version still blows everything else out of water. After using the initial version for several years without updates I'm very happy to see this project alive again! 

It's also nice to see JPEGView published in Winget and Chocolatey repos. Can you add it to Scoop repo as well? In case you haven't heard of it, Scoop is an amazing little package manager, primarily focused on open-source command-line and portable apps (but you can actually install anything with it). The best part is that it installs apps in it's own directory and keeps them "self-contained" within their dirs, including persistent data, dependencies, etc. It's really cool and fits the spirit of software like JPEGView. It would be amazing to be able to install and update JPEGView from Scoop 🍨

Thanks again for your time and effort,
Please keep up the good work!
Kind regards from a dedicated user :-)

**Comment by sylikc** [comment]
> It's also nice to see JPEGView published in Winget and Chocolatey repos. Can you add it to Scoop repo as well? In case you haven't heard of it, Scoop is an amazing little package manager, primarily focused on open-source command-line and portable apps (but you can actually install anything with it). The best part is that it installs apps in it's own directory and keeps them "self-contained" within their dirs, including persistent data, dependencies, etc. It's really cool and fits the spirit of software like JPEGView. It would be amazing to be able to install and update JPEGView from Scoop 🍨

I actually saw in on the Chocolatey repo a few weeks back too.  I didn't post it up there.

I currently post to Winget and PortableApps.  I actually need to think of a way to make sure people are getting the official copies, aka #113

What other repositories would you recommend?

**Reply by sylikc** [reply]
I've submitted a PR to Scoop, but not sure how active the community is... it's been 5 days and not even a single reply... having doubts about Scoop's popularity.

In fact, when I was adding it, there's very slim documentation on Scoop and different pages say different things... so...

**Reply by pureby** [reply]
Thanks for your submission! I don't know much about the process though. I discovered Scoop about half a year ago, and from the user-perspective it's really a great tool, extremely powerful and very intuitive. It's a shame to read that submission process is poorly organized. I hope your app will be added soon :-/

**Reply by sylikc** [reply]
Scoop is really more of a distributed system from what I could tell... but just the process of submission and working with that manifest, I have scathing reviews about it.  **It's barely mature enough for real users to use.**

It has a charter that says it's developer focused, which is pretty much spot on.  I had to Google at least an hour before I figured out how and where to start and how to submit a package... and then the manifest has a whole bunch of undescribed optional fields... the documentation basically says "see this other manifest for an example" https://scoop-docs.vercel.app/docs/concepts/App-Manifests.html#optional-properties ... which is poor documentation, no excuses.

**Reply by pureby** [reply]
Yes. It's really sad to read that submission process is such a mess for the developers. 

And the problem most likely lies within Scoop's popularity indeed, but not for the reason you probably meant. I've done some digging and it seems more like Scoop is becoming _too popular_ for it's current team to sustain. The number of PR's they are receiving last half year seems to be growing significantly — as are the numbers of merged _and_ unprocessed PR's.

What happens is that all new apps have to be reviewed and tested manually by a maintainer (which is a very good requirement for reliability and security), but the problem is that there only seem to be a couple maintainers in the Scoop team. The team, however, is not at all transparent about the way the review process is done, why some packages get approved quickly and other are on hold for weeks, etc. 

It seems like Scoop is suffering from it's "growing pains" this very moment, and has little directive on how to go forward once it totally out-grows it's current maintainer capacity. I hope they will fix this in a timely manner by being more transparent about the ongoing shortcomings of the project and organizing more people to solve those issues (mainly, poorly written documentation and shortage of maintainers).

**Reply by sylikc** [reply]
I'm going to close this discussion... it's been added, but there's no response on the PR... here it is for reference:

https://github.com/ScoopInstaller/Extras/pull/10479 and as per their guidelines, it has to be associated to an issue, which is this one https://github.com/ScoopInstaller/Extras/issues/10478

**Comment by pureby** [comment]
> I currently post to Winget and PortableApps. I actually need to think of a way to make sure people are getting the official copies, aka https://github.com/sylikc/jpegview/issues/113
>
> What other repositories would you recommend?

I am a strong proponent of 20/80 rule: 80% of work can be done with 20% of effort. Therefore I think that publishing official builds to GitHub, Winget, Chocolatey, Scoop and PortableApps will already cover over 90% (if not more) of your potential userbase.

Speaking of Winget, I noticed that JPEGView does not show up when searching for "image", "image viewer", "photo", etc. You can greatly enhance discoverability of your project by adding relevant keywords like `image, images, viewer, viewers, image-viewer, image-viewers, photo, photos, editor, photo-editor, image-editor`, etc. as _tags._ Winget search is still kind of "dumb", so it does not find "viewer" when searching for "viewers" ~~and vice versa~~, this is why I advice to add all tags in plural form as well.

**Reply by sylikc** [reply]
Thanks for the suggestion on `tags`... i've pulled up the whole schema and adding it to the list.  I didn't realize there was so many optional fields in the package manifest!

**Reply by sylikc** [reply]
> winget search image-viewer
```
Name                  Id                           Version     Match             Source
----------------------------------------------------------------------------------------
Image-Viewer          9ND45C9V3ZPW                 Unknown                       msstore
XnViewMP              XnSoft.XnViewMP              1.4.2       Tag: image-viewer winget
XnView                XnSoft.XnView.Classic        2.51.1      Tag: image-viewer winget
ImageView             TonyPottier.ImageView        2.4.6       Tag: image-viewer winget
JPEGView              sylikc.JPEGView              1.1.43      Tag: image-viewer winget
nomacs - Image Lounge nomacs.nomacs                3.16.1709   Tag: image-viewer winget
Quick Picture Viewer  ModuleArt.QuickPictureViewer 3.1.4       Tag: image-viewer winget
qView                 jurplel.qView                5.0         Tag: image-viewer winget
IrfanView             IrfanSkiljan.IrfanView       4.60        Tag: image-viewer winget
ImageGlass            DuongDieuPhap.ImageGlass     8.7.11.6    Tag: image-viewer winget
DjVuLibre DjView      DjVuLibre.DjView             3.5.28+4.12 Tag: image-viewer winget
Honeyview             Bandisoft.Honeyview          5.48        Tag: image-viewer winget
```

It works on the latest 1.1.43! :)

**Comment by pureby** [comment]
Hello @sylikc , just wanted to let you know there is a response from Scoop Maintainer as per 16 april 2023; a request for manifest changes in JPEGView PR: https://github.com/ScoopInstaller/Extras/pull/10479#pullrequestreview-1387008256

Also, the number of unprocessed PR's has gone down significantly from 250+ to 100+ in the last couple of weeks. Somewhere in the discussions it was mentioned that Scoop was suddenly flooded by a very large amount of PR's for new packages, maybe that explains why team struggled to process them in a timely manner. Sometimes a publication in a news article or something similar might have such unpredicted impact (just speculating).
